### DISCUSSION TITLE
How to run the middleware only once at the first visit?

### DISCUSSION THREAD

**Discussion by moaazbhnas228** [vercel/next.js#44717] [Help] [answered]
In a bilingual website, we implemented redirection with the middleware Geolocation object.  
It should only run in the first visit to redirect user to the right locale, and ignore all other in-page navigations (selecting another country). To allow users to checkout other countries.   
In other words it shouldn't run on `<Link />` components or `router.push()`.

The middleware looks like this:

``` js
export async function middleware(req: NextRequest) {
  if (countryFromURL !== countryFromGeolocation) {
    return NextResponse.redirect(countryFromGeolocation);
  }
}
```

There's a dirty solution to add a query param (e.g. `redirection: false`) for all `<Link />`s and `Router.push()`s, but I hope there's a cleaner solution.

**Comment by icyJoseph** [comment, marked as answer]
Perhaps put a cookie on the session? if the cookie is present, skip redirecting.
